internal/db/repositories/user_repository.go

```diff
@@ -4,45 +4,158 @@ package repositories
 
 import (
 	"context"
+	"errors"
+	"strconv"
+	"time"
 
 	"github.com/spf13/viper"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 
 	"github.com/TheDevExperiment/server/internal/db"
+	"github.com/TheDevExperiment/server/internal/log"
+	"github.com/TheDevExperiment/server/internal/utility/funnynamer"
+	"github.com/TheDevExperiment/server/internal/utility/jwt"
 )
 
 type UserRepository struct {
 	collection *mongo.Collection
 }
+type UserModel struct {
+	Id              primitive.ObjectID `bson:"_id,omitempty" `
+	IsGuest         bool               `bson:"isGuest,omitempty"`
+	GuestAuthSecret string             `bson:"guestAuthSecret,omitempty"`
+	CreatedAt       primitive.DateTime `bson:"createdAt,omitempty"`
```

Comment:
we can keep the data type here as int and store unix time here

Reply:
cool, that'll be good!